Meter.c

```diff
@@ -318,25 +325,25 @@ static void LEDMeterMode_draw(Meter* this, int x, int y, int w) {
       y + 2;
    attrset(CRT_colors[LED_COLOR]);
    const char* caption = Meter_getCaption(this);
-   mvaddstr(yText, x, caption);
-   int xx = x + strlen(caption);
+   mvaddnstr(yText, x, caption, w);
```

Comment:
This one is a little beyond my knowledge. According to man pages, the `n` in `mvaddnstr` refers to number of "characters" to print. However, what I really want to limit is the number of terminal columns, not the number of Unicode characters (this distinction is needed if East Asian Wide characters are in the strings).

Perhaps a good way is to test how this function call would output, would it be "`１２`" or just "`１`":

```c
/* "１２３４ fullwidth" */
addnstr("\uFF11\uFF12\uFF13\uFF14 fullwidth", 2);
```

Reply:
What about [`wcswidth(3)`](https://man7.org/linux/man-pages/man3/wcswidth.3.html)?

Reply:
What I wish is a function that does an inverse of [`wcswidth(3)`](https://man7.org/linux/man-pages/man3/wcswidth.3.html). `wcswidth()` returns the terminal column width given a string and a number of (Unicode) characters limit. What I wished is retrieving the number of characters (or number of bytes) given a string and a column width limit. `RichString_appendnWideColumns()` might give the closest I need, but I want a function that operates on a `const char*` rather than a `RichString` object. Maybe I would write one from scratch?

Reply:
Unfortunately that is easier said than actual done, but based on the fact that `wcswidth` returns >=0 you can probably build something like this by calling `wcwidth(3)` for each character in turn until you overshoot the limit, in which case you know the last character was too much (this accounts for zero-width-combiners and other special cases). The down-side is, that you will need to transform the `const char*` into `wchar_t*` first, thus having this functionality live with the other RichString functions might be reasonable.

NB: There's actually an issue that might benefit from this functionality in #462 IIRC.

Reply:
@BenBE Yes, I'm aware of PR #462. And perhaps PR #1231 can benefit, too, as I wish this function would work as a word wrapper that is also CJK-aware.